tests/baselines/reference/typeGuardOfFormTypeOfOther.types

```diff
@@ -63,7 +63,7 @@ else {
     var r2: string | C = strOrC; // string | C

```

Comment:
The comment above in this test says "when false, has not effect on the type of x". You changed this to be "when false, removes classes (objects?) from the type of x", correct?